src/helm/proxy/clients/human_task_client.py

```diff
@@ -0,0 +1,148 @@
+)
+
+from cattrs import unstructure
+
+import surge
+from surge import questions as surge_questions
+
+
+_surge_cache_lock = threading.Lock()
+
+
+class HumanTaskClient(ABC):
+    @abstractmethod
+    def get_human_task(self, request: HumanTaskRequest) -> Optional[HumanTaskRequestResult]:
+        """Get responses to a human task.
+
+        Creates the human task on a labeller platform if it does not already exist.
+        Returns HumanTaskRequestResult if worker answers are complete, or None otherwise.
+        The intended use is to call it once to create the task, wait a while, and then call it
+        later to fetch answers."""
+        pass
+
+
+class RandomHumanTaskClient(HumanTaskClient):
+    """A HumanTaskClient that returns random choices for debugging."""
+
+    def get_human_task(self, request: HumanTaskRequest) -> Optional[HumanTaskRequestResult]:
+        workers: List[HumanTaskWorkerResponse] = []
+        for _ in range(request.template.num_workers):
+            answers: List[str] = []
+            for question in request.template.questions:
+                if question.question_type != "multiple_choice":
+                    raise ValueError("Currently, only multiple_choice questions are supported")
+                answers.append(random.choice(question.options))
+            workers.append(HumanTaskWorkerResponse(answers=answers))
+        return HumanTaskRequestResult(workers=workers)
+
+
+class SurgeAIHumanTaskClient(HumanTaskClient):
+    """A HumanTaskClient that creates tasks for workers on Surge AI."""
+
+    def __init__(self, api_key: str, cache_config: CacheConfig):
+        surge.api_key = api_key
+        self._cache = Cache(cache_config)
+
+    def _to_surge_question(self, question: HumanQuestionTemplate) -> surge_questions.MultipleChoiceQuestion:
+        if question.question_type != "multiple_choice":
+            raise ValueError("Currently, only multiple_choice questions are supported")
+        return surge_questions.MultipleChoiceQuestion(
+            text=question.text,
+            options=question.options,
+        )
+
+    def _get_or_create_surge_project(self, template: HumanTaskTemplate) -> str:
+        """Get or create a project on Surge AI and return the Surge AI project ID.
+
+        Attempt to find a Surge AI project for the template. If one exists, reuse that project.
+        Otherwise, create a new project using the template. Return the Surge AI project ID."""
+
+        def create_surge_project():
+            project = surge.Project.create(
+                name=template.name,
+                instructions=template.instructions,
+                questions=[self._to_surge_question(question) for question in template.questions],
+                num_workers_per_task=template.num_workers,
+            )
+            return {"id": project.id}
+
+        with _surge_cache_lock:
+            project_response, is_cached = self._cache.get({"template": unstructure(template)}, create_surge_project)
```

Comment:
Document precisely what the cache key is

Reply:
Added example.